diff -up mgetty-1.1.36/Makefile.helper2 mgetty-1.1.36/Makefile
--- mgetty-1.1.36/Makefile.helper2	2008-10-09 22:56:45.000000000 +0200
+++ mgetty-1.1.36/Makefile	2008-10-09 22:56:45.000000000 +0200
@@ -623,13 +623,13 @@ install.bin: mgetty sendfax newslock \
 		( mkdir $(spool) && chmod 755 $(spool) )
 	test -d $(FAX_SPOOL) || \
 		( mkdir $(FAX_SPOOL) && \
-		  chown $(FAX_OUT_USER) $(FAX_SPOOL) && \
+		  chown $(FAX_OUT_USER) $(FAX_SPOOL) || : && \
 		  chmod 755 $(FAX_SPOOL) )
 	test -d $(FAX_SPOOL_IN) || \
 		( mkdir $(FAX_SPOOL_IN) && chmod 755 $(FAX_SPOOL_IN) )
 	test -d $(FAX_SPOOL_OUT) || \
 		  mkdir $(FAX_SPOOL_OUT)
-	chown $(FAX_OUT_USER) $(FAX_SPOOL_OUT)
+	chown $(FAX_OUT_USER) $(FAX_SPOOL_OUT) || :
 	chmod 755 $(FAX_SPOOL_OUT)
 #
 # g3 tool programs
